pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -179,17 +179,23 @@ private boolean shouldDeleteInProgressLLCSegment(String segmentName, IdealState
     if (idealState == null) {
       return false;
     }
+    // delete a segment only if it is old enough (5 days) or else,
+    // 1. latest segment could get deleted in the middle of repair by ValidationManager
+    // 2. for a brand new segment, if this code kicks in after new metadata is created but ideal state entry is not yet created (between step 2 and 3),
+    // the latest segment metadata could get marked for deletion
+    if (System.currentTimeMillis() - realtimeSegmentZKMetadata.getCreationTime()
```

Comment:
@npawar  can we have one  time check up front and all the other checks inside that?

Reply:
The time check is only once and we return false if segment is too young to be deleted
All other checks happen only after we come forward from the time check